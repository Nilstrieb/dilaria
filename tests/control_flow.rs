mod common;
use crate::common::_run_test;

run_test!(
    single_if,
    r#"
if true {
    print "true!";
}

if false {
    print "WRONG";
}
"#
);

run_test!(
    if_else,
    r#"
if true {
    print "true!";
} else {
    print "WRONG";
}

if false {
    print "WRONG";
} else {
    print "true!";
}
"#
);

run_test!(
    if_else_if,
    r#"
if false {
    print "WRONG";
} else if true {
    print "true!";
} else {
    print "WRONG";
}

if false {
    print "WRONG";
} else if false {
    print "WRONG";
} else {
    print "true!";
}
"#
);
