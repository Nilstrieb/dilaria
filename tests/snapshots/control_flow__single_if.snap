---
source: tests/control_flow.rs
assertion_line: 4
expression: output

---
"true!\n"
